### ISSUE TITLE
google-auth-library-credentials version collision in dependency tree 

### ISSUE THREAD

**Issue by balopat** [spring-cloud/spring-cloud-gcp#79]
The spring-cloud-gcp-starter-pubsub 1.0.0.BUILD-SNAPSHOT artifact resolves differently in Maven and Gradle (works well in Gradle and breaks in Maven).

The correct version currently seems to be 0.7.0 of google-library-credentials: 
```
        <dependency>
            <groupId>com.google.auth</groupId>
            <artifactId>google-auth-library-credentials</artifactId>
            <version>0.7.0</version>
        </dependency>
```

The maven and gradle dependency resolvers resolve the tree in different order and when the maven one finds the 0.4.0 version of google-authlibrary-credentials (pulled in by io.grpc:grpc-auth:jar:1.4.0:compile). 0.4.0 causes ``java.lang.ClassNotFoundException: com.google.auth.ServiceAccountSigner`` as mentioned [here](https://github.com/spring-guides/getting-started-guides/pull/47#issuecomment-318458869)

See the partial dependency tree dumps below (left is maven right is gradle):

````
[INFO] +- org.springframework.cloud:spring-cloud-gcp-starter-pubsub:jar:1.0.0.BUILD-SNAPSHOT:compile                                    +--- org.springframework.cloud:spring-cloud-gcp-starter-pubsub:1.0.0.BUILD-SNAPSHOT
[INFO] |  +- org.springframework.cloud:spring-cloud-gcp-starter-core:jar:1.0.0.BUILD-SNAPSHOT:compile                                   |    +--- org.springframework.cloud:spring-cloud-gcp-starter-core:1.0.0.BUILD-SNAPSHOT
[INFO] |  |  \- org.springframework.cloud:spring-cloud-gcp-core:jar:1.0.0.BUILD-SNAPSHOT:compile                                        |    |    +--- org.springframework.cloud:spring-cloud-gcp-core:1.0.0.BUILD-SNAPSHOT
[INFO] |  +- org.springframework.cloud:spring-cloud-gcp-pubsub:jar:1.0.0.BUILD-SNAPSHOT:compile                                         |    |    |    +--- com.google.cloud:google-cloud-core:1.2.3
[INFO] |  |  \- org.springframework:spring-messaging:jar:4.3.8.RELEASE:compile                                                          |    |    |    |    +--- com.google.guava:guava:20.0
[INFO] |  +- com.google.cloud:google-cloud-pubsub:jar:0.20.1-beta:compile                                                               |    |    |    |    +--- joda-time:joda-time:2.9.2 -> 2.9.9
[INFO] |  |  +- io.netty:netty-tcnative-boringssl-static:jar:2.0.3.Final:compile                                                        |    |    |    |    +--- org.json:json:20160810 -> 20140107
[INFO] |  |  +- com.google.cloud:google-cloud-core:jar:1.2.1:compile                                                                    |    |    |    |    +--- com.google.api:api-common:1.1.0
[INFO] |  |  |  +- com.google.guava:guava:jar:20.0:compile                                                                              |    |    |    |    |    +--- com.google.code.findbugs:jsr305:3.0.0
[INFO] |  |  |  +- joda-time:joda-time:jar:2.9.9:compile                                                                                |    |    |    |    |    \--- com.google.guava:guava:19.0 -> 20.0
[INFO] |  |  |  +- org.json:json:jar:20140107:compile                                                                                   |    |    |    |    +--- com.google.api:gax:1.4.2
[INFO] |  |  |  +- com.google.api:api-common:jar:1.1.0:compile                                                                          |    |    |    |    |    +--- com.google.auto.value:auto-value:1.2
[INFO] |  |  |  |  \- com.google.code.findbugs:jsr305:jar:3.0.0:compile                                                                 |    |    |    |    |    +--- com.google.guava:guava:20.0
[INFO] |  |  |  +- com.google.api:gax:jar:1.4.1:compile                                                                                 |    |    |    |    |    +--- com.google.auth:google-auth-library-oauth2-http:0.7.0
[INFO] |  |  |  |  +- com.google.auto.value:auto-value:jar:1.2:compile                                                                  |    |    |    |    |    |    +--- com.google.auth:google-auth-library-credentials:0.7.0
[INFO] |  |  |  |  \- org.threeten:threetenbp:jar:1.3.3:compile                                                                         |    |    |    |    |    |    +--- com.google.http-client:google-http-client:1.19.0 -> 1.21.0
[INFO] |  |  |  +- com.google.protobuf:protobuf-java-util:jar:3.3.0:compile                                                             |    |    |    |    |    |    |    +--- com.google.code.findbugs:jsr305:1.3.9 -> 3.0.0
[INFO] |  |  |  |  \- com.google.code.gson:gson:jar:2.8.0:compile                                                                       |    |    |    |    |    |    |    \--- org.apache.httpcomponents:httpclient:4.0.1 -> 4.5.3
[INFO] |  |  |  +- com.google.api.grpc:proto-google-common-protos:jar:0.1.12:compile                                                    |    |    |    |    |    |    |         +--- org.apache.httpcomponents:httpcore:4.4.6
[INFO] |  |  |  \- com.google.api.grpc:proto-google-iam-v1:jar:0.1.12:compile                                                           |    |    |    |    |    |    |         +--- commons-logging:commons-logging:1.2
[INFO] |  |  +- com.google.cloud:google-cloud-core-grpc:jar:1.2.1:compile                                                               |    |    |    |    |    |    |         \--- commons-codec:commons-codec:1.9 -> 1.10
[INFO] |  |  |  +- com.google.protobuf:protobuf-java:jar:3.3.0:compile                                                                  |    |    |    |    |    |    +--- com.google.http-client:google-http-client-jackson2:1.19.0 -> 1.21.0
[INFO] |  |  |  +- io.grpc:grpc-protobuf:jar:1.4.0:compile                                                                              |    |    |    |    |    |    |    +--- com.google.http-client:google-http-client:1.21.0 (*)
[INFO] |  |  |  |  \- io.grpc:grpc-protobuf-lite:jar:1.4.0:compile                                                                      |    |    |    |    |    |    |    \--- com.fasterxml.jackson.core:jackson-core:2.1.3 -> 2.8.8
[INFO] |  |  |  +- io.grpc:grpc-context:jar:1.4.0:compile                                                                               |    |    |    |    |    |    \--- com.google.guava:guava:19.0 -> 20.0
[INFO] |  |  |  \- com.google.api:gax-grpc:jar:0.21.1:compile                                                                           |    |    |    |    |    +--- com.google.api:api-common:1.1.0 (*)
[INFO] |  |  +- com.google.api.grpc:proto-google-cloud-pubsub-v1:jar:0.1.12:compile                                                     |    |    |    |    |    +--- org.threeten:threetenbp:1.3.3
[INFO] |  |  +- com.google.api.grpc:grpc-google-cloud-pubsub-v1:jar:0.1.12:compile                                                      |    |    |    |    |    \--- com.google.code.findbugs:jsr305:3.0.0
[INFO] |  |  +- io.grpc:grpc-netty:jar:1.4.0:compile                                                                                    |    |    |    |    +--- com.google.protobuf:protobuf-java-util:3.3.0 -> 3.3.1
[INFO] |  |  |  +- io.grpc:grpc-core:jar:1.4.0:compile (version selected from constraint [1.4.0,1.4.0])                                 |    |    |    |    |    +--- com.google.protobuf:protobuf-java:3.3.1
[INFO] |  |  |  |  +- com.google.errorprone:error_prone_annotations:jar:2.0.19:compile                                                  |    |    |    |    |    +--- com.google.guava:guava:19.0 -> 20.0
[INFO] |  |  |  |  \- com.google.instrumentation:instrumentation-api:jar:0.4.2:compile                                                  |    |    |    |    |    \--- com.google.code.gson:gson:2.7 -> 2.8.0
[INFO] |  |  |  +- io.netty:netty-codec-http2:jar:4.1.11.Final:compile (version selected from constraint [4.1.11.Final,4.1.11.Final])   |    |    |    |    +--- com.google.api.grpc:proto-google-common-protos:0.1.13
[INFO] |  |  |  |  +- io.netty:netty-codec-http:jar:4.1.11.Final:compile                                                                |    |    |    |    |    +--- com.google.protobuf:protobuf-java:3.3.0 -> 3.3.1
[INFO] |  |  |  |  |  \- io.netty:netty-codec:jar:4.1.11.Final:compile                                                                  |    |    |    |    |    \--- com.google.api:api-common:1.1.0 (*)
[INFO] |  |  |  |  \- io.netty:netty-handler:jar:4.1.11.Final:compile                                                                   |    |    |    |    \--- com.google.api.grpc:proto-google-iam-v1:0.1.13
[INFO] |  |  |  |     \- io.netty:netty-buffer:jar:4.1.11.Final:compile                                                                 |    |    |    |         +--- com.google.protobuf:protobuf-java:3.3.0 -> 3.3.1
[INFO] |  |  |  |        \- io.netty:netty-common:jar:4.1.11.Final:compile                                                              |    |    |    |         +--- com.google.api:api-common:1.1.0 (*)
[INFO] |  |  |  \- io.netty:netty-handler-proxy:jar:4.1.11.Final:compile                                                                |    |    |    |         \--- com.google.api.grpc:proto-google-common-protos:0.1.13 (*)
[INFO] |  |  |     +- io.netty:netty-transport:jar:4.1.11.Final:compile                                                                 |    |    |    \--- org.slf4j:slf4j-api:1.7.25
[INFO] |  |  |     |  \- io.netty:netty-resolver:jar:4.1.11.Final:compile                                                               |    |    +--- com.google.cloud:google-cloud-pubsub:0.20.1-beta
[INFO] |  |  |     \- io.netty:netty-codec-socks:jar:4.1.11.Final:compile                                                               |    |    |    +--- io.netty:netty-tcnative-boringssl-static:2.0.3.Final
[INFO] |  |  +- io.grpc:grpc-stub:jar:1.4.0:compile                                                                                     |    |    |    +--- com.google.cloud:google-cloud-core:1.2.1 -> 1.2.3 (*)
[INFO] |  |  \- io.grpc:grpc-auth:jar:1.4.0:compile                                                                                     |    |    |    +--- com.google.cloud:google-cloud-core-grpc:1.2.1
[INFO] |  |     \- com.google.auth:google-auth-library-credentials:jar:0.4.0:compile                                                    |    |    |    |    +--- com.google.cloud:google-cloud-core:1.2.1 -> 1.2.3 (*)
[INFO] |  +- com.google.cloud:google-cloud-storage:jar:1.2.1:compile                                                                    |    |    |    |    +--- com.google.guava:guava:20.0
[INFO] |  |  +- com.google.cloud:google-cloud-core-http:jar:1.2.1:compile                                                               |    |    |    |    +--- com.google.protobuf:protobuf-java:3.3.0 -> 3.3.1
[INFO] |  |  |  +- com.google.auth:google-auth-library-oauth2-http:jar:0.7.0:compile                                                    |    |    |    |    +--- com.google.protobuf:protobuf-java-util:3.3.0 -> 3.3.1 (*)
[INFO] |  |  |  |  \- com.google.http-client:google-http-client-jackson2:jar:1.19.0:compile                                             |    |    |    |    +--- io.grpc:grpc-protobuf:1.4.0
[INFO] |  |  |  +- com.google.http-client:google-http-client:jar:1.21.0:compile                                                         |    |    |    |    |    +--- io.grpc:grpc-core:1.4.0
[INFO] |  |  |  |  \- org.apache.httpcomponents:httpclient:jar:4.5.3:compile                                                            |    |    |    |    |    |    +--- com.google.guava:guava:19.0 -> 20.0
[INFO] |  |  |  |     +- org.apache.httpcomponents:httpcore:jar:4.4.6:compile                                                           |    |    |    |    |    |    +--- com.google.errorprone:error_prone_annotations:2.0.19
[INFO] |  |  |  |     \- commons-codec:commons-codec:jar:1.10:compile                                                                   |    |    |    |    |    |    +--- com.google.code.findbugs:jsr305:3.0.0
[INFO] |  |  |  +- com.google.oauth-client:google-oauth-client:jar:1.21.0:compile                                                       |    |    |    |    |    |    +--- io.grpc:grpc-context:1.4.0
[INFO] |  |  |  +- com.google.api-client:google-api-client:jar:1.21.0:compile                                                           |    |    |    |    |    |    \--- com.google.instrumentation:instrumentation-api:0.4.2

````

**Comment by joaoandremartins** [comment]
A fix for this will come out in #78

**Comment by joaoandremartins** [comment]
Also, the way I'm planning to address this is to add this as a dependency of spring-cloud-gcp-starter-core. Even though the code itself doesn't require it, any usage does, so I think it makes sense to add it there. I verified the fix works.

There is a newer version btw, 0.7.1.

**Comment by joaoandremartins** [comment]
Should've been fixed on #78, let me know if you're still seeing this.
